tools/make_cjk_ranges.swift

```diff
@@ -0,0 +1,409 @@
+#!/usr/bin/env swift
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2025 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+// Generates a list of CJK character ranges for the purpose of emphasis delimiter calculation.
+// Adapted from `cjk-ranges.ts` in https://github.com/tats-u/markdown-cjk-friendly whose CommonMark
+// specification amendments were used to improve CJK text handling.
+
+import Foundation
+import RegexBuilder
+
+/// The Unicode version used for the following data files. Printed into the generated code file.
+///
+/// > Important: If you update these data files, please change this string and the URLs below.
+/// > Also check the linked section for `unassignedAsCjkRanges` to see if that needs updating.
+let unicodeVersion = "16.0"
+
+// https://www.unicode.org/reports/tr41/tr41-34.html#Data11
+// Local copy fetched from https://www.unicode.org/Public/16.0.0/ucd/EastAsianWidth.txt
+let eawInputDataURL = URL(fileURLWithPath: #file)
+    .deletingLastPathComponent().deletingLastPathComponent()
+    .appending(components: "data", "EastAsianWidth.txt")
```

Comment:
Could avoid some repetition here...

```suggestion
let dataURL = URL(fileURLWithPath: #file)
    .deletingLastPathComponent().deletingLastPathComponent()
    .appending(components: "data")

let eawInputDataURL = dataURL.appending(components: "EastAsianWidth.txt")
```

...and the following paths also.